sympy/calculus/util.py

```diff
@@ -112,3 +119,601 @@ def elm_domain(expr, intrvl):
                                     for element in finite_set])
             _domain = Union(_domain, _domain_element)
         return _domain
+
+
+class Limits(AtomicExpr):
+    """
+    Limits represent an interval `[a, b]`, which is always closed at the
+    ends. Here `a` and `b` can be any value from extended real numbers.
+
+    In many cases it would suffice to know that the limit set is bounded.
+    However, in some other cases more exact information could be useful.
+    For example, that all accumulation values of cos(x) + 1 are non-negative.
+    (Limits(-1, 1) + 1 = Limits(0, 2))
+
+    Let `a` and `b` be reals such that a <= b.
+
+    `\langle a, b\rangle = \{x \in \mathbb{R} \mid a \le x \le b\}`
+
+    `\langle -\infty, b\rangle = \{x \in \mathbb{R} \mid x \le b\}`
```

Comment:
I think that the infinities should be included also here. Both of them, in fact. This is because ``Limits`` do not take into account one-sided limits, by design. Hence ``1/0`` should be regarded as denoting both ``-oo`` and ``oo``` (limits on both sides). Consider, for example, the function ``f = 1/(sin(x) - 1 + 1/x)``:
```
>>> limit(1/(sin(x) - 1 + 1/x), x, oo)
<-oo, -1/2>
```
The intended meaning of ``Limits`` is to give an approximate location of the accumulation points of a real function at a limit point. In this case we can also take the points ``x_n = pi/2 + 2*n*pi`` for which we have ``f(x_n) = x_n`` with limit ``oo``. Hence ``oo`` should be included in ``Limits``. This does not mean that any computations need be redefined. This is purely a question of how ``Limits`` are to be interpreted. (We can think of ``-oo`` and ``oo`` as being identified similarly as when they are mapped to ``zoo``.)